src/unstoppableAPI.ts

```diff
@@ -117,7 +141,21 @@ export default class Udapi extends NamingService {
 
   /** @ignore */
   private findMethod(domain: string) {
-    return [new Zns(), new Ens()].find(m => m.isSupportedDomain(domain));
+    try {
+      const method = this.findMethodOrThrow(domain);
+      return method;
+    } catch (err) {
+      if (err instanceof ResolutionError) return null;
+      throw err;
+    }
+  }
+
+  private findMethodOrThrow(domain: string) {
+    const method = [new Zns(), new Ens()].find(m =>
+      m.isSupportedDomain(domain),
+    );
+    if (!method) throw new ResolutionError('UnsupportedDomain', { domain });
+    return method;
```

Comment:
It is cleaner to call `findMethod` from `findMethodOrThrow` in this particular case.